### DISCUSSION TITLE
Manual control over toot translations

### DISCUSSION THREAD

**Discussion by toonvandenbos** [mastodon/mastodon#20894] [General]
The Mastodon instance I'm part of recently added automatic translations to toots. It's great!

However, as an author it would be helpful to also have full manual control over a selection of important locales.

[Multilingual countries](https://en.wikipedia.org/wiki/List_of_multilingual_countries_and_regions) like Belgium (we speak dutch, french and german) have a multilingual audience. When publishing posts, we know our followers will prefer or need an adapted version for their mother tongue. Therefore, it is quite common to translate content into the main official languages. 

In most cases automated translations would be fine, but even when done right, they don't always keep the stylistic and humoristic traits of the original content. Furthermore, language-based audiences are often linked to communities with strong identities, in which case it's important to use an adapted tone of voice or rhetorical approach.

Of course, all this could be true for international audiences too, especially when authors know their posts will target a few linguistic communities in particular.

Most of the other social media websites I had the opportunity to use don't handle this aspect correctly, resulting for instance in posts beginning with "_[english version below]_", which greatly misses the author's intended communication strategy. In my opinion, it should be possible to add manual translations when I want a specific version for a certain locale and let the automated translations handle the others.

If Mastodon could cover this feature, it would be a real game changer compared to mainstream social media sites.

**Comment by yelvington** [comment]
I enabled translation on my server this morning, and I'm not seeing the behavior you describe. It appears that if translation is enabled, a "Translate" link is offered when the post locale does not match the user locale. Translation does not happen until the user clicks the link.  

![screenshot-s yelvington com-2022 11 30-08_58_32](https://user-images.githubusercontent.com/157818/204832233-c608cd33-b0db-4b48-a997-82d71eb9fb6d.png)


If someone posts in DE but has the locale set to EN, I don't get the translate link. (Post setting is right below the text input box in the web UI.)

**Reply by toonvandenbos** [reply]
The feature I'm talking about does not concern the "Translate" link below the post directly. It is a suggestion aiming to improve the result of post translations: as an author, I'd want to be able to provide manual translations for some specific locales. The goal is to avoid automatic translations when interacting with known multilingual audiences and let the automatic translations cover the other, unspecified translations.

**Comment by Nobodysbetter** [comment]
A possible implementation could be to use LANGUAGE or LOCATION as a FLAG like adding colors inline to chat. 

[US/ENGLISH]//Hello\\
[DE/GERMAN]//Guten Tag\\

--Nobody
